scripts/pasha_main.coffee

```diff
@@ -246,9 +267,22 @@ module.exports = (robot) ->
             util.pagerdutyAlert("outage: #{pashaState.prio1.title}")
             scribeLog "confirmed prio1"
             robot.receive(new TextMessage(msg.message.user,
-                "#{botName} changelog addsilent #{user} confirmed the prio1"))
+              "#{botName} changelog addsilent #{user} confirmed the prio1"))
+
+            channelName = "prio1-#{dateformat(new Date(), 'yyyy-mm-dd')}"
+            util.slackApi("channels.create", {name: channelName, token: constant.slackApiNonbotToken}, (err, res, data) ->
+                if !err && data.ok
+                    pashaState.prio1.channel = {id: data.channel.id, name: channelName}
+                    robot.brain.set(constant.pashaStateKey, JSON.stringify(pashaState))
+                    msg.send("Created channel ##{channelName}, please join and keep all prio1 communication there.")
+                    invitePrio1RolesToPrio1SlackChannel()
+                else
+                    msg.send("Failed to create channel #{channelName}: #{err || data.error}")
+
+            )
+
```

Comment:
That's a good one, open an issue to track it?

On Thu, Feb 11, 2016 at 9:48 PM Bence Kiglics <notifications@github.com>
wrote:

> In scripts/pasha_main.coffee
> <https://github.com/prezi/pasha/pull/14#discussion_r52664347>:
>
> > @@ -246,9 +267,22 @@ module.exports = (robot) ->
> >              util.pagerdutyAlert("outage: #{pashaState.prio1.title}")
> >              scribeLog "confirmed prio1"
> >              robot.receive(new TextMessage(msg.message.user,
> > -                "#{botName} changelog addsilent #{user} confirmed the prio1"))
> > +              "#{botName} changelog addsilent #{user} confirmed the prio1"))
> > +
> > +            channelName = "prio1-#{dateformat(new Date(), 'yyyy-mm-dd')}"
> > +            util.slackApi("channels.create", {name: channelName, token: constant.slackApiNonbotToken}, (err, res, data) ->
> > +                if !err && data.ok
> > +                    pashaState.prio1.channel = {id: data.channel.id, name: channelName}
> > +                    robot.brain.set(constant.pashaStateKey, JSON.stringify(pashaState))
> > +                    msg.send("Created channel ##{channelName}, please join and keep all prio1 communication there.")
> > +                    invitePrio1RolesToPrio1SlackChannel()
> > +                else
> > +                    msg.send("Failed to create channel #{channelName}: #{err || data.error}")
> > +
> > +            )
> > +
>
> Possible improvement for the future: archive it automatically after x
> hours of silence :).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/prezi/pasha/pull/14/files#r52664347>.
>
-- 
*"Abesto" Zoltan Nagy*
Tools Engineer at Prezi <http://prezi.com/>